src/linalg.jl

```diff
@@ -88,6 +88,17 @@ function svd_blockwise(B::BlockDiagonal{T}; full::Bool=false) where T
     end
     return BlockDiagonal(U), S, BlockDiagonal(Vt)
 end
+function svd_blockwise(B::BlockDiagonal{T, <:Tuple}; full::Bool=false) where T
+    S = Vector{float(T)}()
+    U_Vt = ntuple(length(blocks(B))) do i
+        F = svd(getblock(B, i), full=full)
+        append!(S, F.S)
+        (F.U, F.Vt)
```

Comment:
How come we aren't `push!`ing like above? Or vice versa if `first.()`/`last.()` is better?

Reply:
because above we want `U` to be a `Vector`, so create a Vector and add the elements. Whereas here we want it to be a `Tuple` hence `ntuple`. 

rather than `push!` we should probably be preallocating a Vector of the right length and overwriting the values, but the two functions would still be different (since `Tuple`s are immutable)